src/external/smartsheet.ts

```diff
@@ -0,0 +1,135 @@
+import createHttpError from 'http-errors';
+import { StatusCodes } from 'http-status-codes';
+import fetch from 'node-fetch';
+
+import { smartsheetId, smartsheetToken } from '../config/env';
+import config from '../config/project';
+import { IUserOuput } from '../db/models/User';
+
+const parseRoles = (roles: string[]) =>
+    roles.map((role) => config.roleOptions.find((option) => option.value === role)?.label || role).join(', ');
+
+export const handleNewsletterUpdate = async (user: IUserOuput): Promise<any> => {
+    switch (user.newsletter_subscription_status) {
+        case 'subscribed':
+            await subscribeNewsletter(user);
+            break;
+        case 'unsubscribed':
+            await unsubscribeNewsletter(user.newsletter_email);
+            break;
+        default:
+            break;
+    }
+};
+
+export const subscribeNewsletter = async (user: IUserOuput): Promise<any> => {
+    const url = `https://api.smartsheet.com/2.0/sheets/${smartsheetId}/rows`;
```

Comment:
everything's specific to include's project. Clin uses users-api and doesn't have any newsletter for now, let's imagine it's having one but no smartsheet, we'll have to do breaking chances here to handle multiple newsletter provider. Can we think of a way to handle this case ?